src/service.rs

```diff
@@ -524,13 +523,26 @@ impl Service {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn start<S: AsRef<OsStr>>(&self, service_arguments: &[S]) -> Result<()> {
+    pub fn start<S: AsRef<OsStr>>(&self, service_arguments: &[S]) -> Result<(), Error> {
         let wide_service_arguments = service_arguments
             .iter()
-            .map(|s| WideCString::from_str(s).chain_err(|| ErrorKind::InvalidStartArgument))
-            .collect::<Result<Vec<WideCString>>>()?;
-        let mut raw_service_arguments: Vec<*const u16> =
-            wide_service_arguments.iter().map(|s| s.as_ptr()).collect();
+            /*
+            .map(|s| match WideCString::from_str(s) {
+                Err(_) => Err(ErrorKind::InvalidStartArgument.into()),
+                Ok(s) => Ok(s),
+            })
+            */
+            .map(|s| WideCString::from_str(s).map_err(|e| e.into()))
+            .collect::<Vec<Result<WideCString, Error>>>();
```

Comment:
Isn't it more convenient to collect into `Result<Vec<WideCString>, Error>` the way we used to?

Disclaimer: Haven't done rust in a while.

Reply:
i cleaned this up a bit, slightly more verbose than with `error-chain`

Reply:
changed the collection pattern to more closely resemble what it was with `error-chain`